src/hashring.c

```diff
@@ -86,10 +86,16 @@ size_t hashring_size(hashring_t ring) {
 void* hashring_choose(struct hashring *ring,
 		      const char *key,
 		      uint32_t *shard_num) {
+	if (ring == NULL) {
+		return NULL;
+	}
+
```

Comment:
Need to check if backends is null, one crash happens because backends is NULL...